kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/convert/BigQuerySchemaConverter.java

```diff
@@ -116,6 +120,35 @@ public BigQuerySchemaConverter(boolean allFieldsNullable) {
     return com.google.cloud.bigquery.Schema.of(fields);
   }
 
+  private void throwOnCycle(Schema kafkaConnectSchema, List<Schema> seenSoFar) {
```

Comment:
It's probably more performant to do this check as we convert the schema instead of doing it separately as a pre-flight check, but this shouldn't incur a large performance penalty since it's only performed on table schema updates, which shouldn't occur very frequently.

(It's likely you've already thought this through. Just commenting here for posterity's sake 🙂 )

Reply:
Yeah, the code would end up being less clean and touch all the convertX() methods to pipe the seen list through. So, I opted to do the check out-of-band. Like you said, performance shouldn't take a huge hit. If so, we are doing something wrong.